### PR TITLE
Handle pax install

### LINKED ISSUE #253: Dependency complaints from zigi when upgrading zopen
I've been seeing these complaints for a few weeks now when upgrading `zopen`.  I don't have zigi installed, and yet I see error messages about it.

```
Checking dependencies for zigi: git...
installDependencies 15: installPort 45: installPorts 8: /u/home/ANGIO/meta/bin/lib/zopen-install 454: FSUM7343 cannot open "zigi/.depsenv" for output: EDC5129I No such file or directory.
installDependencies 27: installPort 45: installPorts 8: /u/home/ANGIO/meta/bin/lib/zopen-install 454: FSUM7343 cannot open "zigi/.depsenv" for output: EDC5129I No such file or directory.
Checking git...
New release with tag "gitport_783" found for gitport
Preparing to download git
Downloading latest release from gitport...
***WARNING: Could not download https://github.com/ZOSOpenTools/gitport/releases/download/gitport_783/git-2.39.2.20230320_172306.zos.pax.Z
installDependencies 33: installPort 45: installPorts 8: /u/home/ANGIO/meta/bin/lib/zopen-install 454: FSUM7343 cannot open "zigi/.depsenv" for output: EDC5129I No such file or directory.
```

### LINKED ISSUE #381: Make zopen install way less noisy
Have the default be to just show what is being installed and any errors or warnings. Use -v for all the other stuff... 

For example, a consumer installing bash sees:
image

but probably all they need is the yellow and green lines (and if other tools installed, show them too, but I think they are either yellow or green). 

since I said -y to questions, I don't think I would care about disk usage...

@IgorTodorovskiIBM @DevonianTeuchter @HarithaIBM curious what you think...

### LINKED ISSUE #468: Add support when there is no tty available
We use stty to get the columns value, but stty fails because there is not tty available. We shouldn't need a tty when requesting an install

From @ccw-1 
```
- Querying for installed packages
stty: !FSUMB045 failed to get termios attributes: EDC5137I Inappropriate I/O control operation.
printInstalledEntries 3: /home/ccw/meta/bin/zopen-query 380: FSUM6659 syntax error in arithmetic expression " / 4 -1 " near /
- Querying remote repo for latest package information
- ```

### LINKED ISSUE #472: Installing a package and dependencies prompts the user multiple times
I installed the `vim` package on a new zopen install for the first time, and `zopen` asked me if I was sure.  When I replied `y`, the install continued.  After it was complete, `zopen` asked me if I wanted to install `ncurses`.  Since this is a dependency, it should have been called out at the start of the install, and listed with the initial prompt.  Otherwise installing a package with a long tail of dependencies will require the user to baby-sit the installation script, answering yes to each question in turn.

### LINKED ISSUE #533: Can we optimize "Checking N potential links?"
This takes quite a while, especially for vim.
```
- Checking for obsoleted files in /home/itodoro/zopen/usr/ tree from git
- Checking 604 potential links
```
Can we store the links in a cache or database and remove them?

### PR DESCRIPTION
**Pull request by DevonianTeuchter**
Handle installing direct from pax file;
- move "install" code from zopen-install into common.sh to enable creation of zopen-upgrade script [separation of concerns!] with appropriate parameters added/removed from each module
- refactoring monolithic "install" code [now in common.sh] into more specialized functions 
Numerous qol fixes found while smoke testing including: 
 - maintaining installation history so dependencies aren't checked multiple times (resolves #472  #431)
 - allowing an "a" option for the "_Do you want to continue (y/n/a)?_" question to automatically answer "yes" to any future questions [in case a command is run without -y/--yes and the user actually meant to!
 - performance when alternating/upgrading improved by diffing updated files rather than systematically checking all symlinks for existence [still required for removal]; resolve #533 as much as possible [removal processing will always be slow due to the need to check every possible file a package ships to see if that symlink is still active]
 - improving query of installed packages
 - improving screen formatting calculation using screen cols (resolve #468)
 - make install less noisier still  (resolve #381)
 - resolve #253 

A global configuration for a paxfile repository [currently a single directory] can be set during initialization using the `--paxrepodir <dir>` option. This can be manually adjusted in the global config at <ZOPEN_ROOTFS>/etc/zopen/paxrepodir. ` --paxrepodir` can also be used with `zopen install ` to force a lookup of that repo for the specified package/pax.

Paxes can be installed using the command:
`zopen install --paxinstall path/to/<paxfile.pax.Z>`   - installs pax as found at specified location
`zopen install --paxinstall <package>`   - attempts to locate a pax for package in the current direcotry or in the global paxrepodir if configured
`zopen install --paxinstall --paxrepodir <dir> <package>.pax.Z` - install the specified package pax file, looking in <dir> for the file

Note that the manual has not been updated - the code is semi-"dark-shipped" for those able to test while the normal installation methodology continues as normal.  Once stabilised, the official documentation can be updated.

### CODE REVIEW

**Review by MikeFultonDev — CHANGES_REQUESTED** [review]
Please add some tests to the framework and ensure no regressions on existing tests before we merge

File: bin/zopen-init [hunk 1]
```diff
@@ -41,6 +42,12 @@ Options:
                     globally configure the release line for package
                     installs to enable Development (DEV) packages; the
                     default is for a system to use STABLE packages
+      --paxrepodir <RPEO>
```
Comment (IgorTodorovskiIBM) [inline bin/zopen-init:45]: duplicate flags? Also, RPEO -> REPO

**Comment by IgorTodorovskiIBM** [comment]
I'm seeing some changes that are probably not intended. 
You may need to do a git rebase --interactive to remove some of the commits not intended in this PR (like this https://stackoverflow.com/questions/36168839/how-to-remove-commits-from-a-pull-request)

**Comment by DevonianTeuchter** [comment]
Additional testing added under https://github.com/ZOSOpenTools/metaport/pull/28

**Comment by DevonianTeuchter** [comment]
Should also resolve: https://github.com/ZOSOpenTools/metaport/issues/23

**Review by MikeFultonDev — APPROVED** [review]
did not mean to approve - ignore.

**Review by MikeFultonDev — COMMENTED** [review]
Hm. just a draft. ignore my approval 😁

**Pull request closed without merging**